ci/do_ci.sh

```diff
@@ -17,6 +17,42 @@ function install_prometheus_cpp_client
   popd
 }
 
+function run_benchmarks
+{
+  docker run -d --rm -it -p 4317:4317 -p 4318:4318 -v \
+    $(pwd)/examples/otlp:/cfg otel/opentelemetry-collector:0.38.0 \
+    --config=/cfg/opentelemetry-collector-config/config.dev.yaml
+
+  [ -z "${BENCHMARK_DIR}" ] && export BENCHMARK_DIR=$HOME/benchmark
+  mkdir -p $BENCHMARK_DIR
+  bazel $BAZEL_STARTUP_OPTIONS build $BAZEL_OPTIONS -c opt -- \
+    $(bazel query 'attr("tags", "benchmark_result", ...)')
+  echo ""
+  echo "Benchmark results in $BENCHMARK_DIR:"
+  (
+    cd bazel-bin
+    find . -name \*_result.json -exec bash -c \
+      'echo "$@" && mkdir -p "$BENCHMARK_DIR/$(dirname "$@")" && \
+       cp "$@" "$BENCHMARK_DIR/$@" && chmod +w "$BENCHMARK_DIR/$@"' _ {} \;
+  )
+
+  # collect benchmark results into one array
+  pushd $BENCHMARK_DIR
+  components=(api sdk exporters)
+  for component in "${components[@]}"
+  do
+    out=$component-benchmark_result.json
+    find ./$component -type f -name "*_result.json" -exec cat {} \; > $component_tmp_bench.json
+    cat $component_tmp_bench.json | docker run -i --rm itchyny/gojq:0.12.6 -s \
```

Comment:
nit - one minor comment, if it's not a major rework, can we use [jq](https://stedolan.github.io/jq/download/) instead of `gojq`, it's more lightweight without any external dependencies.

Reply:
I've tried jq, it was showing different behavior on my local (Ubuntu 20.4) and on the GHA vm (Ubuntu latest). Collecting the benchmark results of all the tests into one array didn't work on the jq of GHA vm. So I switched to gojq which has a docker image.